### PR TITLE
Yuri/eng 1110 resolving json scalars handle plain types

### PR DESCRIPTION
**Pull request by YuriBuerov**
Addressed the following issues:
- https://github.com/wundergraph/wundergraph/issues/448
-  https://github.com/wundergraph/wundergraph/issues/562

### CODE REVIEW

**Comment by jensneuse** [comment]
@YuriBuerov sorry but I feel a bit lost. Can you explain in simple terms how this solves the 2 linked issues?
I think we've come to a place where the functions used in `Load()` might need some comments and descriptions to make sense to an outsider. Why do we use which function and in which case? Would be nice if we could make this a bit more clear.

File: pkg/engine/resolve/resolve.go [hunk 1]
```diff
@@ -952,6 +953,13 @@ func (r *Resolver) resolveString(ctx *Context, str *String, data []byte, stringB
 
 	if str.UnescapeResponseJson {
 		value = bytes.ReplaceAll(value, []byte(`\"`), []byte(`"`))
+
+		// if value is a string
```
Comment (devsergiy) [inline pkg/engine/resolve/resolve.go]: could we extend comment to smth like that?

```go
// When the original value from usptream response was a string `"hello"`
// after getting it via jsonparser.Get we will get unquoted value `hello`,
// which is invalid json and we need to quote it again
```

File: pkg/engine/resolve/resolve_test.go [hunk 2]
```diff
@@ -1514,6 +1514,106 @@ func TestResolver_ResolveNode(t *testing.T) {
 			},
 		}, Context{Context: context.Background()}, `{"id":1,"name":"Jens","pet":{"name":"Woofie"}}`
 	}))
+	t.Run("json object within a string", testFn(false, false, func(t *testing.T, ctrl *gomock.Controller) (node Node, ctx Context, expectedOutput string) {
```
Comment (devsergiy) [inline pkg/engine/resolve/resolve_test.go]: we could add an additional wrapping test group "with unescape json enabled"

File: pkg/engine/resolve/resolve_test.go [hunk 3]
```diff
@@ -1514,6 +1514,106 @@ func TestResolver_ResolveNode(t *testing.T) {
...
+					HasBuffer: true,
+				},
+			},
+		}, Context{Context: context.Background()}, `{"data":[1, 2, 3]}`
+	}))
+	t.Run("json boolean within a string", testFn(false, false, func(t *testing.T, ctrl *gomock.Controller) (node Node, ctx Context, expectedOutput string) {
+		return &Object{
+			Fetch: &SingleFetch{
+				BufferId:   0,
+				DataSource: FakeDataSource(`{"data": "true"}`),
```
Comment (devsergiy) [inline pkg/engine/resolve/resolve_test.go]: hmm, looks like an edge case

upstream is returning a string `"true"` not a boolean `true`
wdt should we treat it as a string?

Comment (devsergiy) [reply pkg/engine/resolve/resolve_test.go]: disregard, it is desired with unescapeJson

Comment (YuriBuerov) [reply pkg/engine/resolve/resolve_test.go]: I think we need to be consistent. If we treat an object encoded in a string as a JSON object, I would expect similar behavior for boolean/array etc. What do you think?

File: pkg/engine/resolve/resolve_test.go (hunk 2 above)
Comment (YuriBuerov) [reply pkg/engine/resolve/resolve_test.go]: it's enabled, check line 1530, is that what you mean?

**Review by spetrunin — APPROVED** [review]

**Comment by greemo** [comment]
Hi guys, thanks for the effort on this. Is there a way to disable the unescapeResponseJson functionality?

We're using Wundergraph in a very generic platform, and unescaping a JSON response will not always be appropriate.

For me, it would work best as part of the data source configuration. I've looked through the code and could not find any config to enable/disable it. Please excuse me if it's there, and point me in the right direction.

Cheers!

**Comment by spetrunin** [comment]
@greemo Could I ask you to open a discussion/issue in the Wundergraph repo and describe your use case?

**Pull request merged**